datalad/support/gitrepo.py

```diff
@@ -846,6 +845,23 @@ def get_file_content(self, file_, branch='HEAD'):
             return content_str.splitlines()
         # TODO: keep splitlines?
 
+    def get_files_history(self, files, branch='HEAD'):
+        """
+
+        Parameters
+        ----------
+        files: list
+          list of files, only commits with queried files are considered
+        branch: str
+          Name of the branch to query. Default: HEAD.
+
+        Returns
+        -------
+        [iterator]
+        yielding Commit items generator from branch history associated with files
+        """
+        return gitpy.objects.commit.Commit.iter_items(self.repo, branch, paths=files)
```

Comment:
No it doesn't seem to include merge commits. The commit on which the file was actually edited and commited is the ones in which are listed